runtime/vm/vmprops.c

```diff
@@ -1433,3 +1445,651 @@ getMUtf8String(J9JavaVM *vm, const char *userString, UDATA stringLength)
+	if (NULL != envSpace) {
+		j9mem_free_memory(envSpace);
+	}
+
+failed:
+	if (NULL != homeAlloc) {
+		j9mem_free_memory(homeAlloc);
+	}
+	return plist;
+}
+
+#endif /* defined(WIN32) */
+
+/**
+ * @brief Get the default system properties.
+ *
+ * @return an Object array of system property key value pairs
+ */
+jobjectArray
+getSystemPropertyList(JNIEnv *env)
+{
+	PORT_ACCESS_FROM_ENV(env);
+	int propIndex = 0;
+	jobject propertyList = NULL;
+#define PROPERTY_COUNT 137
+#if JAVA_SPEC_VERSION < 17
+	char *propertyKey = NULL;
+	const char * language = NULL;
+	const char * region = NULL;
+	const char * variant = NULL;
+#endif /* JAVA_SPEC_VERSION < 17 */
+	const char *strings[PROPERTY_COUNT] = {0};
+#define USERNAME_LENGTH 128
+	char username[USERNAME_LENGTH] = {0};
+	char *usernameAlloc = NULL;
+	/* buffer to hold the size of the maximum direct byte buffer allocations */
+	char maxDirectMemBuff[24] = {0};
+	IDATA result = 0;
+
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *javaVM = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = javaVM->internalVMFunctions;
+
+	/* Change the allocation value PROPERTY_COUNT above as you add/remove properties,
+	 * then follow the propIndex++ convention and consume 2 * slots for each property. 2 * number of property keys is the
+	 * correct allocation.
+	 * Also note the call to addSystemProperties below, which may add some configuration-specific properties.  Be sure to leave
+	 * enough room in the property list for all possibilities.
+	 */
+
+	if (J9_GC_POLICY_METRONOME == javaVM->omrVM->gcPolicy) {
+		strings[propIndex++] = "com.ibm.jvm.realtime";
+		strings[propIndex++] = "soft";
+	}
+
+#if defined(J9VM_OPT_SHARED_CLASSES)
+	strings[propIndex++] = "com.ibm.oti.shared.enabled";
+	if ((NULL != javaVM->sharedClassConfig)
+		&& J9_ARE_ALL_BITS_SET(javaVM->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHE_NON_BOOT_CLASSES)
+	) {
+		strings[propIndex++] = "true";
+	} else {
+		strings[propIndex++] = "false";
+	}
+#endif
+
+	strings[propIndex++] = "ibm.signalhandling.sigchain";
+	if (J9_ARE_ANY_BITS_SET(javaVM->sigFlags, J9_SIG_NO_SIG_CHAIN)) {
+		strings[propIndex++] = "false";
+	} else {
+		strings[propIndex++] = "true";
+	}
+	strings[propIndex++] = "ibm.signalhandling.sigint";
+	if (J9_ARE_ANY_BITS_SET(javaVM->sigFlags, J9_SIG_NO_SIG_INT)) {
+		strings[propIndex++] = "false";
+	} else {
+		strings[propIndex++] = "true";
+	}
+
+	/* The JCLs use ibm.signalhandling.rs to determine if they should prevent the registration of signal handlers for what
+	 * 	we consider to be asynchronous signals.
+	 * The JCLs do not install handlers for any synchronous signals */
+	strings[propIndex++] = "ibm.signalhandling.rs";
+	if (J9_ARE_ALL_BITS_SET(javaVM->sigFlags, J9_SIG_XRS_ASYNC)) {
+		strings[propIndex++] = "true";
+	} else {
+		strings[propIndex++] = "false";
+	}
+
+	strings[propIndex++] = "com.ibm.vm.bitmode";
+#ifdef J9VM_ENV_DATA64
```

Comment:
Please use `#if defined(X)` instead of `#ifdef X` and add the missing comments (lines 1961-1980).